src/main/java/org/springframework/data/jpa/repository/config/JpaRepositoryConfigExtension.java

```diff
@@ -234,9 +237,17 @@ public void registerBeansForRoot(BeanDefinitionRegistry registry, RepositoryConf
 	@Override
 	protected ClassLoader getConfigurationInspectionClassLoader(ResourceLoader loader) {
 
-		ClassLoader classLoader = loader.getClassLoader();
-
-		return classLoader != null && LazyJvmAgent.isActive(loader.getClassLoader())
+        AtomicReference<ClassLoader> classLoaderRef = new AtomicReference<>();
+        AccessController.doPrivileged(new PrivilegedAction() {
```

Comment:
`AccessController` is subject to deprecation and future removal with the deprecation of Java's Security Manager.

Reply:
will revert related changes then